core/architecture.md

```diff
@@ -0,0 +1,150 @@
+chain for those pods. Remember, the chain may be updated at anytime and include
+new or different SFC classification or CNFs, a key advantage of Nimbess.
+
+UNP is able to classify traffic from L2-L7. The goal of UNP is to offer service
+policy mesh as used with other applications like [Istio](https://istio.io)
+without the need for an Envoy sidecar proxy application. The goal is to
+consolidate this type of functionality in the high speed dataplane, which will
+give better performance and a single fabric for all network data traffic.
+
+## Networking in Nimbess
+
+Most CNIs today offer L3 CNI solutions where some form of routing is used to
+control traffic between pods, especially across nodes. Some of these CNIs use
+static routing, their own routing implementation, or well known routing
+protocols such as BGP, which offers high scalability, fast route convergence
+times, as well as features like Equal-Cost Multipath (ECMP), Anycast, and
+VPN support. Therefore for L3 networking, Nimbess leverages iBGP as the control
+plane protocol for routing traffic between pods. Alternatively, Nimbess also 
+supports full L2 fabric across the cluster. This is important as many CNFs
+are L2 and may require a full L2 fabric to function correctly.
+
+## Architecture
+
+Nimbess consists of several components ran as containers across Kubernetes:
+
+[![Nimbess Architecture](../img/Nimbess.png)](../img/Nimbess.png)
+
+### Nimbess ETCD
+The distributed data store used by Nimbess in order to maintain k8s resources,
+as well as per data plane and network fabric configuration.
+
+### Nimbess Stargazer
+Stargazer is responsible for watching k8s resources and reflecting them into
+the Nimbess etcd. Stargazer also handles delegating work to Nimbess Agents if
+further processing is required for those resources.
+
+### Nimbess Agent
+Nimbess Agent acts as the distributed controller across the k8s deployment. The
+Agent is responsible for translating k8s resources into network resources.
+While any Agent is capable of performing this function, the actual work of
+rendering network resources into data plane configuration, as well as
+configuring the data plane is only performed by the data plane's local Agent.
+Each Agent has a full view of the network and all other Agents. The Agent is
+also responsible for watching BGP routes injected in the kernel and configuring
+the data plane appropriately.
+
+### Nimbess CNI
+Invoked by Kubelet and communicates with Nimbess Agent in order to handle CNI
+calls.
+
+### Calico BIRD/Felix Agent
+Calico is used to provide the BGP stack and route insertion into the kernel.
+
+### Climate Control
+Climate Control serves as the CLI tool for interacting directly with Nimbess in
+order to examine data plane state and Nimbess specific configuration.
+
+### Doppler
+Doppler provides a customized GUI in order to visually examine the k8s cluster
+network, as well as visualize the data plane network fabric.
```

Comment:
Any option to modify configuration or some troubleshooting functions like initiating packet traces.

Reply:
That's a good idea. We have that capability to tap anywhere in any BESS pipeline. I'll reword this.